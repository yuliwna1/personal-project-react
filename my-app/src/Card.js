import React from "react";

function Card(props) {
  return (
    <a
      href={`https://github.com/${props.link}`}
      target="_blank"
      className="card"
    >
      <p className="bold">{props.name}</p>

      <p>
        <span>{props.type === "ForkEvent" ? "Forked from: " : "Status: "}</span>
        {props.info}
      </p>
    </a>
  );
}

export default Card;
