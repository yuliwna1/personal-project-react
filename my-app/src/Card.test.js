import React from "react";
import Card from "./Card";
import renderer from "react-test-renderer";

const props = {
  name: "yuliwna1",
  type: "ForkEvent"
};

test("Card component rendered", () => {
  const component = renderer.create(<Card></Card>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
