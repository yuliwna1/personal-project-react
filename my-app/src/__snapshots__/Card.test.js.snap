// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component rendered 1`] = `
<a
  className="card"
  href="https://github.com/undefined"
  target="_blank"
>
  <p
    className="bold"
  />
  <p>
    <span>
      Status: 
    </span>
  </p>
</a>
`;
